In [1]:
# Importing necessary libraries

import pandas as pd
import numpy as np

In [2]:
# Reading the given 'star_craft_player' dataset

player_df = pd.read_csv("dataset/data/starcraft_player_data.csv")

In [3]:
# Looking deeply into the dataset

player_df.shape

(3395, 20)

After looking at the shape attribute of the dataset, we understood that there are over 3395 rows and 20 columns in the dataset.

In [4]:
# Finding the first 10 rows of the dataset

player_df.head(10)

GameID  LeagueIndex Age HoursPerWeek TotalHours       APM  SelectByHotkeys  \
0      52            5  27           10       3000  143.7180         0.003515   
1      55            5  23           10       5000  129.2322         0.003304   
2      56            4  30           10        200   69.9612         0.001101   
3      57            3  19           20        400  107.6016         0.001034   
4      58            3  32           10        500  122.8908         0.001136   
5      60            2  27            6         70   44.4570         0.000978   
6      61            1  21            8        240   46.9962         0.000820   
7      72            7  17           42      10000  212.6022         0.009040   
8      77            4  20           14       2708  117.4884         0.002944   
9      81            4  18           24        800  155.9856         0.005054   

   AssignToHotkeys  UniqueHotkeys  MinimapAttacks  MinimapRightClicks  \
0         0.000220              7        0.000110            0.000392   
1         0.000259              4        0.000294            0.000432   
2         0.000336              4        0.000294            0.000461   
3         0.000213              1        0.000053            0.000543   
4         0.000327              2        0.000000            0.001329   
5         0.000255              2        0.000000            0.000000   
6         0.000169              6        0.000000            0.000045   
7         0.000676              6        0.001164            0.001253   
8         0.000527              2        0.000019            0.000414   
9         0.000524              8        0.000025            0.000399   

   NumberOfPACs  GapBetweenPACs  ActionLatency  ActionsInPAC  \
0      0.004849         32.6677        40.8673        4.7508   
1      0.004307         32.9194        42.3454        4.8434   
2      0.002926         44.6475        75.3548        4.0430   
3      0.003783         29.2203        53.7352        4.9155   
4      0.002368         22.6885        62.0813        9.3740   
5      0.002425         76.4405        98.7719        3.0965   
6      0.001988         94.0227        90.5311        4.1017   
7      0.004952         24.6117        41.7671        6.6104   
8      0.005399         52.0140        46.4321        3.3746   
9      0.003569         24.4632        52.1538        6.5664   

   TotalMapExplored  WorkersMade  UniqueUnitsMade  ComplexUnitsMade  \
0                28     0.001397                6          0.000000   
1                22     0.001193                5          0.000000   
2                22     0.000745                6          0.000000   
3                19     0.000426                7          0.000000   
4                15     0.001174                4          0.000000   
5                16     0.000372                6          0.000000   
6                15     0.000573                5          0.000000   
7                45     0.002277                9          0.000129   
8                29     0.001035                7          0.000273   
9                27     0.001310                6          0.000000   

   ComplexAbilitiesUsed  
0              0.000000  
1              0.000208  
2              0.000189  
3              0.000384  
4              0.000019  
5              0.000000  
6              0.000000  
7              0.000249  
8              0.000470  
9              0.000000

GameID represents the id of the particular game which means it is the unique identifier, LeagueIndex is the id of the unique league this is the target variable as it can be used to predict player's rank. Going forward, we can try to eliminate any features based on the correlation with target variable, or check for the importance by analyzing the features description.

In [5]:
# Looking at the structure of the dataset

str(player_df)

'      GameID  LeagueIndex Age HoursPerWeek TotalHours       APM  \\\n0         52            5  27           10       3000  143.7180   \n1         55            5  23           10       5000  129.2322   \n2         56            4  30           10        200   69.9612   \n3         57            3  19           20        400  107.6016   \n4         58            3  32           10        500  122.8908   \n...      ...          ...  ..          ...        ...       ...   \n3390   10089            8   ?            ?          ?  259.6296   \n3391   10090            8   ?            ?          ?  314.6700   \n3392   10092            8   ?            ?          ?  299.4282   \n3393   10094            8   ?            ?          ?  375.8664   \n3394   10095            8   ?            ?          ?  348.3576   \n\n      SelectByHotkeys  AssignToHotkeys  UniqueHotkeys  MinimapAttacks  \\\n0            0.003515         0.000220              7        0.000110   \n1            0.003304         0

In [13]:
# Finding out the summary of the dataset

player_df.describe()

GameID  LeagueIndex          APM  SelectByHotkeys  \
count   3395.000000  3395.000000  3395.000000      3395.000000   
mean    4805.012371     4.184094   117.046947         0.004299   
std     2719.944851     1.517327    51.945291         0.005284   
min       52.000000     1.000000    22.059600         0.000000   
25%     2464.500000     3.000000    79.900200         0.001258   
50%     4874.000000     4.000000   108.010200         0.002500   
75%     7108.500000     5.000000   142.790400         0.005133   
max    10095.000000     8.000000   389.831400         0.043088   

       AssignToHotkeys  UniqueHotkeys  MinimapAttacks  MinimapRightClicks  \
count      3395.000000    3395.000000     3395.000000         3395.000000   
mean          0.000374       4.364654        0.000098            0.000387   
std           0.000225       2.360333        0.000166            0.000377   
min           0.000000       0.000000        0.000000            0.000000   
25%           0.000204       3.000000        0.000000            0.000140   
50%           0.000353       4.000000        0.000040            0.000281   
75%           0.000499       6.000000        0.000119            0.000514   
max           0.001752      10.000000        0.003019            0.004041   

       NumberOfPACs  GapBetweenPACs  ActionLatency  ActionsInPAC  \
count   3395.000000     3395.000000    3395.000000   3395.000000   
mean       0.003463       40.361562      63.739403      5.272988   
std        0.000992       17.153570      19.238869      1.494835   
min        0.000679        6.666700      24.093600      2.038900   
25%        0.002754       28.957750      50.446600      4.272850   
50%        0.003395       36.723500      60.931800      5.095500   
75%        0.004027       48.290500      73.681300      6.033600   
max        0.007971      237.142900     176.372100     18.558100   

       TotalMapExplored  WorkersMade  UniqueUnitsMade  ComplexUnitsMade  \
count       3395.000000  3395.000000      3395.000000       3395.000000   
mean          22.131664     0.001032         6.534021          0.000059   
std            7.431719     0.000519         1.857697          0.000111   
min            5.000000     0.000077         2.000000          0.000000   
25%           17.000000     0.000683         5.000000          0.000000   
50%           22.000000     0.000905         6.000000          0.000000   
75%           27.000000     0.001259         8.000000          0.000086   
max           58.000000     0.005149        13.000000          0.000902   

       ComplexAbilitiesUsed  
count           3395.000000  
mean               0.000142  
std                0.000265  
min                0.000000  
25%                0.000000  
50%                0.000020  
75%                0.000181  
max                0.003084

From this we can say that the highest rank in the league is 8, lowest is one. In the league, we can also observe that the least action per minute is equal to 22, and the maximum action per minute is 390. The number of 24x24 game coordinate grids viewed by the player per timestamp that is the 'TotalMapExplored' is ranging from 5 to 58 and other features are as represented above.